searchcore/src/vespa/searchcore/proton/matching/extract_features.h

```diff
@@ -25,7 +25,7 @@ struct ExtractFeatures {
      * documents (must be in ascending order) using unpack information
      * from a search.
      **/
-    static FeatureSet::UP get_feature_set(SearchIterator &search, RankProgram &rank_program, const std::vector<uint32_t> &docs);
+    static FeatureSet::UP get_feature_set(const MatchToolsFactory &mtf, SearchIterator &search, RankProgram &rank_program, const std::vector<uint32_t> &docs);
```

Comment:
prefer passing the rename map instead of the full match tools factory to reduce coupling

Reply:
done